### PR TITLE
added template_kinds controller to api controllers

### PR DESCRIPTION
**Pull request by dmitri-d**

### CODE REVIEW

File: app/controllers/api/v1/template_kinds_controller.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+module Api
+  module V1
+    class TemplateKindsController < V1::BaseController
+
+      api :GET, "/template_kinds/", "List all template kinds."
+      param :page, String, :desc => "paginate results"
+      param :per_page, String, :desc => "number of entries per request"
```
Comment (skottler) [inline app/controllers/api/v1/template_kinds_controller.rb:7]: Should this really be passed as a string? Seems a more like an int.

**Comment by skottler** [comment]
Looks good other than the small comment about the `per_page` parameter. @isratrade, can you take a look as well?

**Comment by skottler** [comment]
Also, the tests pass - http://ci.theforeman.org/job/test_develop_pull_request/18/.

**Comment by isratrade** [comment]
@skottler, you are right that :page and :per_page are integers, but all parameters are passed as strings.  If I change  type to Integer, it causes a validation error in apipie.  See example below.  @iNecas, can you comment on this?

$ curl -u admin:secret -H 'accept:application/json' http://0.0.0.0:3000/api/architectures?page=1

{
  "error": {
    "class": "Apipie::ParamInvalid",
    "message": "Invalid parameter 'page' value \"1\": Must be Integer",
    "parameter_name": "page"
  }
}

**Comment by ohadlevy** [comment]
@isratrade on a seperate topic, I dont think we need to report the class as part of the validation, not sure if its configurable from apipie point of view /cc @iNecas

File: app/controllers/api/v1/template_kinds_controller.rb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+      def index
+        @template_kinds = TemplateKind.all.paginate(paginate_options)
```
Comment (ohadlevy) [inline app/controllers/api/v1/template_kinds_controller.rb:10]: .all.paginate is usually wrong (as it would convert it to an array first)

**Comment by ohadlevy** [comment]
amended and merged - thanks @witlessbird

**Pull request closed without merging**